src/main/java/com/thealgorithms/scheduling/NonPreemptivePriorityScheduling.java

```diff
@@ -0,0 +1,116 @@
+package com.thealgorithms.scheduling;
+
+import java.util.PriorityQueue;
+
+/**
+ * This class implements the Non-Preemptive Priority Scheduling algorithm.
+ * Processes are executed in order of their priority. The process with the
+ * highest priority (lower priority number) is executed first,
+ * and once a process starts executing, it cannot be preempted.
+ */
+public final class NonPreemptivePriorityScheduling {
+
+    private NonPreemptivePriorityScheduling() {
+    }
+
+    /**
+     * Represents a process with an ID, burst time, priority, and start time.
+     */
+    static class Process implements Comparable<Process> {
+        int id;
+        int startTime;
```

Comment:
I meant not the time when the process is started, but when it arrives in the system. It appears and can be executed

Reply:
I added another variable called `arrivalTime` to keep track of this.